legion/legion/k8s/services.py

```diff
@@ -351,6 +352,38 @@ def scale(self, new_scale):
 
         self.reload_cache()
 
+    def check_service_is_deleted(self):
+        """
+        Check that this service has been deleted
+
+        :return: bool -- has services been deleted
+        """
+        client = legion.k8s.utils.build_client()
+        core_v1api = kubernetes.client.CoreV1Api(client)
+
+        actual_services = [
+            item.metadata.name
+            for item
+            in core_v1api.list_namespaced_service(self.namespace).items
+        ]
+        return self.k8s_service.metadata.name not in actual_services
+
+    def check_deployment_is_deleted(self):
+        """
+        Check that this deployment has been deleted
+
+        :return: bool -- has deployment been deleted
+        """
+        client = legion.k8s.utils.build_client()
+        extension_api = kubernetes.client.ExtensionsV1beta1Api(client)
+
+        actual_deployments = [
+            item.metadata.name
+            for item
+            in extension_api.list_namespaced_deployment(self.namespace).items
```

Comment:
@kirillmakhonin add retries for extension_api.list_namespaced_deployment(self.namespace)